File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        def nCr(n,r):
            ans = 1
            for i in range(r):
                ans = ans * (n-i)
                ans = ans / (i+1)
            return round(ans)

        res = []
        for i in range(1, numRows+1):
            temp = []
            for j in range(1,i+1):
                temp.append(nCr(i - 1, j - 1))
            res.append(temp)
        return res

File: 0118-pascals-triangle/0118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
    let res = []
    var ncr = function (n, row) {
        // if (row > n) return 0;
        // if (row === 0 || row === n) return 1;
        let res = 1
        for (let i = 0; i < row; i++) {
            res = res * (n - i)
            res = res / (i + 1)
        }
        return Math.round(res)
    }
    for (let i = 1; i <= numRows; i++) {
        let temp = []
        for (let j = 1; j <= i; j++) {
            temp.push(ncr(i - 1, j - 1))
        }
        res.push(temp)
    }
    return res
};